docs/debugging-transactions.md

```diff
@@ -0,0 +1,510 @@
+        - [With `opcodeLogger`](#2-with-opcodeLogger)
+
+## Overview
+
+[Remix IDE](https://remix-project.org/) provides powerful debugging tools that allow developers to inspect and analyze
+the execution of smart contracts.
+
+This guide will walk you through the various debugging capabilities available, including:
+
+1. how to use the built-in debugger in Remix to step through the execution of a transaction and inspect the state of
+   your smart contracts, and
+2. how to manually call the `debug_traceTransaction` method and analyze the response to get detailed information about
+   the execution of a transaction.
+
+## 1. Debugging with Remix
+
+The built-in debugger in Remix allows you to step through the execution of a transaction, inspect the state, and
+understand the behavior of your smart contracts.
+
+### Steps to Use the Built-in Debugger
+
+Official Documentation: [Remix Debugger](https://remix-ide.readthedocs.io/en/latest/debugger.html)
+
+1. **Set Up Remix with a Local Node**:
+    - Ensure you have a local node running that supports the `debug_traceTransaction` method.
+    - Connect Remix to your local node by selecting
+      the ["External Http Provider"](https://remix-ide.readthedocs.io/en/latest/run.html#more-about-external-http-provider)
+      option in the Remix IDE and entering the URL of your local node (e.g., `http://localhost:7546`).
+
+2. **Deploy or Load Contract**:
+    - [Create and deploy your smart contract](https://remix-ide.readthedocs.io/en/latest/create_deploy.html) using Remix
+      or load an existing contract by providing its address.
+
+3. **Run a Transaction**:
+    - Interact with your deployed contract by sending a transaction (e.g., calling a function).
+
+4. **Access the Debugger**:
+    - Remix displays information related to each transaction result in the terminal.
+    - After the transaction has been processed, check in the terminal to see where it is logged.
+    - Find the transaction you just executed and click on the debug icon next to it.
+
+5. **Use the Debugger**:
+    - The Remix Debugger will open, allowing you to step through the transaction execution.
+    - You can inspect the state, stack, memory, any errors and other details at each step of the execution.
+
+### Example of Debugging a Transaction
+
+Here’s a simple example of how to deploy and interact with a contract in Remix:
+
+#### Solidity Contract
+
+```solidity
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.0;
+
+contract SimpleStorage {
+    uint256 public storedData;
+
+    function set(uint256 x) public {
+        storedData = x;
+    }
+
+    function get() public view returns (uint256) {
+        return storedData;
+    }
+}
+```
+
+#### Steps in Remix
+
+1. **Deploy the Contract**:
+    - Compile the `SimpleStorage` contract.
+    - Deploy it using the "Deploy & Run Transactions" tab.
+
+2. **Interact with the Contract**:
+    - Call the `set` function with a value (e.g., `set(42)`).
```

Comment:
Would a screenshot help explain this?  Or a direct link to some remix documentation page that describes calling a contract method?

Reply:
I did the contract deploy and function call manually in the Remix IDE through MetaMask and then changed the provider to “External HTTP provider” to be able to call the `debug_traceTransaction` cause MetaMask does not support this method.

I can describe this in those bullet points, thanks for the feedback!